tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -86,7 +86,10 @@ public static void Initialize()
 
             LifetimeManager.Instance.AddAsyncShutdownTask(ShutdownAsync);
 
-            TracerSettings tracerSettings = _settings.TracerSettings;
+            var tracerSettings = _settings.TracerSettings;
+
+            // We disable obfuscation of the query string (is not required by CI Visibility and allocates 16MB due the custom Regex implementation)
+            tracerSettings.ObfuscationQueryStringRegex = string.Empty;
```

Comment:
I'll try lazy loading this in another PR instead of removing the support. Restoring original code in https://github.com/DataDog/dd-trace-dotnet/pull/3809/commits/bd0c36061f2029d42ef292d23c7555468a369b65

Reply:
I did the lazy loading as part of this PR here https://github.com/DataDog/dd-trace-dotnet/pull/3809/commits/04304228a0c03a66267343b03938c6401e673c35